feat(apple): Note on Carthage for Apple Silicon (#2996)
getsentry/sentry-docs: src/platforms/apple/common/usage.mdx

- We recommend upgrading to 6.x.x to get reliable <PlatformLink to="/configuration/releases/">Release Health</PlatformLink> data,
as we added many improvements over 5.x.x, in which we introduced Release Health.
- If you're currently using a version between 6.0.2 and 6.0.8, we **highly recommend** you upgrade to the latest version to avoid [a regression](https://github.com/getsentry/sentry-cocoa/pull/786) that affected the quality of the stacktraces and error messages, which was fixed in 6.0.9.

> - For using Sentry via Carthage on Macs with Apple Silicon, please use Carthage >= 0.37.0, which introduced support for XCFramework needed for Apple Silicons. Please read the [release notes of Carthage 0.37.0](https://github.com/Carthage/Carthage/releases/tag/0.37.0) for more details.

Reviewer: ```suggestion
- To use Sentry via Carthage on Macs with Apple Silicon, please use Carthage >= 0.37.0, which introduced support for XCFramework needed for Apple Silicons. Learn more in the [release notes for Carthage 0.37.0](https://github.com/Carthage/Carthage/releases/tag/0.37.0).
```
